docs/standards.md

```diff
@@ -0,0 +1,52 @@
+# Mapping Standards
+
+This document details the standards that all contributions to Parchment mappings must follow.
+
+### Parameter Names
+
+1. Use American English for spellings
+    - Example: `color`, not `colour`, `armor`, not `armour`
+2. All names should be prefixed with `p`
+    - Example: `pLevel`, `pXPos`
+3. Names should make sense without the prefix, and should be valid java identifiers NOT reserved keywords as such:
+    - Example: `pFloat` is NOT valid, as `float` is a reserved keyword.
+    - Example: `pOs` is not a correct use of the prefix, it should be `pPos`
+4. Parameter names may be named based on the types, and the context in which they are used. They should be verbose and use complete words - do not omit essential information for brevity's sake.
+    - Example: `BlockPos pAdjacentPos, BlockPos pCurrentPos`, not `BlockPos pPos1, BlockPos pPos2`
+5. Avoid single letters, or abbreviations.
+    - Example: `pNorthWest` over `pNW`, `pMatrixStack` over `pMStack`. 
+    - Exception: common abbreviations can be used: `IO` / `NBT`, etc.
+6. Use Lower Camel Case (`lowerCamelCase`)
+7. Do not use `$` or `_` in variable names
+8. All parameters should match the regex `p[A-Z][A-Za-z0-9]+`.
+    - Example: `pDot`, `pPos`, `pBlockState`
+9. Favor using Mojang / Mojmap class names over <1.17 MCP classes. In general, the mojmap name is the de-facto standard (some exceptions may apply).
+    - Example: Use `level` over `world`, `blockEntity` over `tileEntity`
+10. Common parameter names can use abbreviations for brevity:
+    - Example: `BlockPos pos, Level level, BlockState state, WorldGenLevel level`
+
+
+A note about lambda parameters: They can, and will conflict with both existing variables, other methods, and other lambdas. Care should be taken to review these, until such a time that an automated and/or testable solution is in place to verify that these cannot conflict.  
+
+### Javadocs
+
+Prioritization:
+
+- Prioritize documenting public APIs over private fields and methods.
+- Prioritize methods and parameters, important fields, and classes.
+- Follow the principle of least effort for best payoff!
+- Do not document `package-info.java`.
+
+Content:
+
+1. Fully qualified mojmap class names should be used in javadoc tags such as `@link`.
+    - Example: `{@code net.minecraft.math.BlockPos}` instead of `{@code BlockPos`}
```

Comment:
Based on https://github.com/ParchmentMC/mappings-testing/pull/1#discussion_r645109509, we should adjust the content information to include the same.